runelite-client/src/main/java/net/runelite/client/plugins/StatusBars/StatusBarsOverlay.java

```diff
@@ -0,0 +1,324 @@
+	private static final Color VENOMED_COLOR = new Color(0, 65, 0, 150);
+	private static final Color HEAL_COLOR = new Color (255, 112, 6, 150);
+	private static final Color PRAYER_HEAL_COLOR = new Color(57, 255, 186, 75);
+	private static final BufferedImage NO_IMAGE = new BufferedImage(17, 17, BufferedImage.TYPE_INT_ARGB);
+	private static final int HEIGHT = 252;
+	private static final int WIDTH = 20;
+	private static final int PADDING = 1;
+	private static final int IMAGE_SIZE = 16;
+	private static final int HEALTH_LOCATION_X = 2;
+	private static final int PRAYER_LOCATION_X = 3;
+	private final Client client;
+	private final StatusBarsConfig config;
+	private final SkillIconManager skillIconManager;
+	private final QueryRunner queryRunner;
+	private final ItemStatChanges itemStatChanges;
+	private final TextComponent textComponent = new TextComponent();
+
+	@Inject
+	private StatusBarsOverlay(Client client, StatusBarsConfig config, SkillIconManager skillIconManager, QueryRunner queryRunner, ItemStatChanges itemStatChanges)
+	{
+		setPosition(OverlayPosition.DYNAMIC);
+		setLayer(OverlayLayer.ABOVE_WIDGETS);
+		this.client = client;
+		this.config = config;
+		this.skillIconManager = skillIconManager;
+		this.queryRunner = queryRunner;
+		this.itemStatChanges = itemStatChanges;
+	}
+
+	@Override
+	public Dimension render(Graphics2D g)
+	{
+		int offsetHealthX = 0;
+		int offsetPrayerX  = 0;
+		int offsetHealthY = 0;
+		int offsetPrayerY = 0;
+		int height = 0;
+
+		// gets the current viewport and changes values accordingly so the bars are in the right position.
+		for (Viewport viewport : Viewport.values())
+		{
+			Widget viewportWidget = client.getWidget(viewport.getViewport());
+			if (viewportWidget != null && !viewportWidget.isHidden())
+			{
+				height = HEIGHT;
+				Point offsetLeft = viewport.getOffsetLeft();
+				Point offsetRight = viewport.getOffsetRight();
+				Point location = viewportWidget.getCanvasLocation();
+				offsetPrayerX = (location.getX() - offsetRight.getX());
+				offsetHealthX = (location.getX() - offsetLeft.getX());
+				offsetHealthY = (location.getY() - offsetLeft.getY());
+				offsetPrayerY = (location.getY() - offsetRight.getY());
+
+				if (viewport == Viewport.RESIZED_BOTTOM)
+				{
+					offsetPrayerY = (location.getY() - 18 - offsetRight.getY());
+					offsetHealthY = (location.getY() - 18 - offsetRight.getY());
+					height += 20;
+				}
+				else
+				{
+					offsetPrayerX = (location.getX() + viewportWidget.getWidth() - offsetRight.getX());
+				}
+			}
+		}
+
+		// hides the bars when inventory is hidden in resizable and when the bank is open.
+		Widget widgetBankTitleBar = client.getWidget(WidgetInfo.BANK_TITLE_BAR);
+		Widget bottomLine = client.getWidget(WidgetInfo.RESIZABLE_VIEWPORT_BOTTOM_LINE_INTERFACE_CONTAINER);
+		if (widgetBankTitleBar != null)
+		{
+			return null;
+		}
+
+		if (client.isResized() && client.getWidget(WidgetInfo.RESIZABLE_VIEWPORT_BOTTOM_LINE_INTERFACE_CONTAINER) != null)
```

Comment:
This seems like duplicated check? You already have the widget in the Viewport class

Reply:
If the bottomline interface is hidden it's null and throws lot's of nullPointerExceptions because it can't get the is.Hidden() value because it's "null", without that check the plugin gets flooded with NullPointerExceptions. this was the only way i was able to make it work